src/main/java/org/blondin/mpg/AbstractClient.java

```diff
@@ -53,10 +76,67 @@ protected void setProxy(Proxy proxy) {
     }
 
     protected <T> T post(String path, MultivaluedMap<String, Object> headers, Object entityRequest, Class<T> entityResponse, boolean wrapRoot) {
-        return call(path, headers, entityRequest, entityResponse, wrapRoot);
+        return call(path, headers, entityRequest, entityResponse, wrapRoot, -1);
     }
 
-    private <T> T call(String path, MultivaluedMap<String, Object> headers, Object entityRequest, Class<T> entityResponse, boolean wrapRoot) {
+    private <T> T call(String path, MultivaluedMap<String, Object> headers, Object entityRequest, Class<T> entityResponse, boolean wrapRoot,
+            long cacheTimeMilliSecond) {
+        try {
+            if (StringUtils.isBlank(url)) {
+                throw new UnsupportedOperationException("Please use 'setUrl(...)' before using this client");
+            }
+            System.out.println(String.format("--- DEBUG: Call on: %s (cacheTime: %s)", url + path, cacheTimeMilliSecond));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/axel3rd-github/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.blondin%3Ampg-coach-bot&pullRequest=28&issues=AWb1cGKxsSodgw_KSkaz&open=AWb1cGKxsSodgw_KSkaz)

[ISSUE_KEY]: AWb1cGKxsSodgw_KSkaz